src/ir/wireable.cpp

```diff
@@ -27,17 +27,23 @@ Wireable::~Wireable() {
   }
 }
 
+Select* Wireable::make_sel(const std::string& sel_str, Type* type, bool is_instance) {
+  Select* select = new Select(this->getContainer(), this, sel_str, type, is_instance);
+  selects[sel_str] = select;
+  return select;
+}
+
 Select* Wireable::sel(const std::string& selStr) {
   if (selects.count(selStr)) {
     return selects[selStr];
   }
+  if (isa<Instance>(this) &&
```

Comment:
I think self is a little awkward, maybe instA._instances_ or some other special but explicit keyword. Using self is confusing w.r.t. it referencing the interface in one context and instances in another context.

Is having the same name for an instances and a port desirable? I guess it’s technically okay because interface ports are prefixed by self

Reply:
I’ll move the logic into instance::sel (I originally did this but I think I ran into some issue, but I’ll reply if I come across it again)

Reply:
I think _instances_ is also slightly semantically wrong as well. We want a word that is saying "select the inside of this current instance". Also something concise would be nice for serializing long hierarchical select paths.

Reply:
Maybe a separator different than dot?

Reply:
I would be fine with that. Semicolon perhaps.

"instA;in0" vs "instA.in0"
and for the select path:
("instA", ";", "in0") vs ("instA", "in0")

Reply:
updated it to use `;` for instances